sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2267,6 +2267,215 @@ case class FormatNumber(x: Expression, d: Expression)
   override def prettyName: String = "format_number"
 }
 
+/**
+ * A function that converts string to numeric.
```

Comment:
This comments duplicates the `usage` part. I don't think it is useful.

Reply:
I will preserve this line and change the `usage` part as `Convert `strExpr` to a number based on the `patternExpr`.`